### ISSUE TITLE
Make example executors part of library?

### ISSUE THREAD

**Issue by rmlarose** [unitaryfund/mitiq#282]
Since the [examples](https://github.com/unitaryfund/mitiq/blob/master/docs/source/guide/guide_03-executors.rst) are written with arbitrary observables, these are really quite useful, and we might want to consider making them part of the library.

Although Mitiq is sort of middleware, it would be nice (especially for us) to not have to think about how Qiskit simulation works and just do

```python
circuit = my_qiskit_circuit
obs = my_numpy_array

executor = partial(mitiq_qiskit.qs_noisy_sampling_sim, obs=obs, shots=1000, noise=0.1)

mitiq.execute_with_zne(executor, circuit)
```

We could generalize the noisy executors to input the type of noise as well as the noise level. And could potentially do a Qiskit qpu backend, although this may be more complicated. Regardless, people can still define their own executor, so we don't lose anything here.

Thoughts?

**Comment by willzeng** [comment]
These are useful. They strike me as something that the different libraries themselves should provide. Could we upstream these executors into their respective libraries?

That could be cleanest in terms of library modularity and maintenance. 

Then we'd just update the docs with nice one-liner executors (as it should be :) )

**Comment by karalekas** [comment]
I think having a library of executors would be a great idea (that's what I started to do in #339 for pyquil/QCS). I also agree with you @willzeng, that it would be nice if these existed upstream (and thus were maintained by others) but I think that might be a tough sell (we'd essentially be asking people to maintain a separate API for us). Thus, I don't think it's unreasonable to start by putting the executors in mitiq now and then trying to get third-party buy-in later.

**Comment by rmlarose** [comment]
I believe we agreed that example executors should be part of the library. #339 has been merged and #379 is specifically for Qiskit executors.

Closing this on account of the agreement and newer, more specific issues (i.e., this is now superseded by #379).
